File: Backend/TrackrAPI/TrackrAPI/Dtos/Seguridad/AccesoMenuDto.cs
using System.Collections.Generic;

namespace TrackrAPI.Dtos.Seguridad
{
    public class AccesoMenuDto
    {
        public int IdAcceso { get; set; }
        public int? IdAccesoPadre { get; set; }
        public string Nombre { get; set; }
        public string Clave { get; set; }
        public string Url { get; set; }
        public string ClaseIcono { get; set; }
        public string UrlImagen { get; set; }
        public List<AccesoMenuDto> Hijos { get; set; }
        public string ClaveRolAcceso { get; set; }
        public string ClaveTipoAcceso { get; set; }
    }
}

File: Frontend/trackr-administrador/src/app/shared/dtos/seguridad/acceso-menu-dto.ts
export class AccesoMenuDto {
  public idAcceso: number;
  public nombre: string;
  public clave: string;
  public url: string;
  public claseIcono: string;
  public urlImagen: string;
  public hijos: AccesoMenuDto[];
  public claveRolAcceso: string;
  public claveTipoAcceso: string;
}
